etc/kernelspecs/dask_python_yarn_remote/scripts/launch_ipykernel.py

```diff
@@ -0,0 +1,285 @@
+    while not shutdown:
+        request = get_gateway_request(sock)
+        if request:
+            signum = -1  # prevent logging poll requests since that occurs every 3 seconds
+            if request.get('signum') is not None:
+                signum = int(request.get('signum'))
+                os.kill(os.getpid(), signum)
+            elif request.get('shutdown') is not None:
+                    shutdown = bool(request.get('shutdown'))
+            if signum != 0:
+                logger.info("gateway_listener got request: {}".format(request))
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument('connection_file', nargs='?', help='Connection file to write connection info')
+    parser.add_argument('--RemoteProcessProxy.response-address', dest='response_address', nargs='?',
+                        metavar='<ip>:<port>', help='Connection address (<ip>:<port>) for returning connection file')
+    parser.add_argument('--RemoteProcessProxy.kernel-id', dest='kernel_id', nargs='?',
+                        help='Indicates the id associated with the launched kernel.')
+    parser.add_argument('--RemoteProcessProxy.port-range', dest='port_range', nargs='?',
+                        metavar='<lowerPort>..<upperPort>', help='Port range to impose for kernel ports')
+
+    arguments = vars(parser.parse_args())
+    connection_file = arguments['connection_file']
+    response_addr = arguments['response_address']
+    kernel_id = arguments['kernel_id']
+    lower_port, upper_port = _validate_port_range(arguments['port_range'])
+    ip = "0.0.0.0"
+
+    if connection_file is None and kernel_id is None:
+        raise RuntimeError("At least one of the parameters: 'connection_file' or "
+                           "'--RemoteProcessProxy.kernel-id' must be provided!")
+
+    if kernel_id is None:
+        logger.warning("Parameter 'connection_file' is deprecated.  Update kernel.json file to use "
+                       "'--RemoteProcessProxy.kernel-id {kernel_id}'")
+
+    # If the connection file doesn't exist, then create it.
+    if (connection_file and not os.path.isfile(connection_file)) or kernel_id is not None:
+        key = str_to_bytes(str(uuid.uuid4()))
+        connection_file = determine_connection_file(connection_file, kernel_id)
+
+        ports = _select_ports(5, lower_port, upper_port)
+
+        write_connection_file(fname=connection_file, ip=ip, key=key, shell_port=ports[0], iopub_port=ports[1],
+                              stdin_port=ports[2], hb_port=ports[3], control_port=ports[4])
+        if response_addr:
+            gateway_socket = return_connection_info(connection_file, response_addr, lower_port, upper_port)
+            if gateway_socket:  # socket in use, start gateway listener thread
+                gateway_listener_thread = Thread(target=gateway_listener, args=(gateway_socket,))
+                gateway_listener_thread.start()
+
+    namespace = initialize_namespace()
```

Comment:
Here's the other portion of [dynamic config determination](https://github.com/jupyter/enterprise_gateway/blob/master/etc/kernel-launchers/python/scripts/launch_ipykernel.py#L314-L324).